dataframe/field_config.go

```diff
@@ -0,0 +1,162 @@
+package dataframe
+
+import (
+	"encoding/json"
+)
+
+// FieldConfig represents the display properties for a Field.
+type FieldConfig struct {
+
+	// This struct needs to match the frontend component defined in:
+	// https://github.com/grafana/grafana/blob/master/packages/grafana-data/src/types/dataFrame.ts#L23
+	// All properties are optional should be omitted from JSON when empty or not set.
+
+	Title      string     `json:"title,omitempty"`
+	Filterable Filterable `json:"filterable,omitempty"` // indicates if the Field's data can be filtered by additional calls.
+
+	// Numeric Options
+	Unit     string   `json:"unit,omitempty"`     // is the string to display to represent the Field's unit, such as "Requests/sec"
+	Decimals *uint16  `json:"decimals,omitempty"` // is the number of decimal places to display
+	Min      *float64 `json:"min,omitempty"`      // is the maximum value of fields in the column. When present the frontend can skip the calculation.
+	Max      *float64 `json:"max,omitempty"`      // see Min
+
+	// Convert input values into a display string
+	Mappings []ValueMapping `json:"mappings,omitempty"`
+
+	// Map numeric values to states
+	Thresholds *ThresholdsConfig `json:"thresholds,omitempty"`
+
+	// Map values to a display color
+	// NOTE: this interface is under development in the frontend... so simple map for now
+	Color map[string]interface{} `json:"color,omitempty"`
+
+	// Used when reducing field values
+	NullValueMode NullValueMode `json:"nullValueMode,omitempty"`
+
+	// The behavior when clicking on a result
+	Links []DataLink `json:"links,omitempty"`
+
+	// Alternative to empty string
+	NoValue string `json:"noValue,omitempty"`
+
+	// Panel Specific Values
+	Custom map[string]interface{} `json:"custom,omitempty"`
+}
+
+// Filterable is a tri-state bool (unset(nil)/false/true)
+type Filterable *bool
+
+// FilterableTrue returns s Filterable set to True
+func FilterableTrue() Filterable {
+	return Filterable(&([]bool{true}[0]))
+}
+
+// FilterableFalse returns a Filterable set to False
+func FilterableFalse() Filterable {
+	return Filterable(&([]bool{false}[0]))
+}
```

Comment:
Rather than a specific "Filterable" type, does it make more sense to have a "NullableFlag" or "NullableBool" -- that way we have the field ready for other types that will get added like
```
	Filterable NullableFlag `json:"filterable,omitempty"` 
	Sortable  NullableFlag `json:"sortable,omitempty"` 
```

Reply:
Used chained setter pattern instead so it is like the other pointer things on the struct.